### PR TITLE
OCPBUGS-21499: update golang.org/x/net to v0.17.0

### PR DESCRIPTION
**Pull request by zshi-redhat**
fix CVE-2023-39325

Not a cherry-pick, manually run go get

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -50,14 +50,14 @@ require (
...
 	golang.org/x/oauth2 v0.0.0-20220411215720-9780585627b5 // indirect
-	golang.org/x/sys v0.1.0 // indirect
-	golang.org/x/term v0.1.0 // indirect
-	golang.org/x/text v0.4.0 // indirect
+	golang.org/x/sys v0.13.0 // indirect
+	golang.org/x/term v0.13.0 // indirect
+	golang.org/x/text v0.13.0 // indirect
 	golang.org/x/time v0.0.0-20220411224347-583f2d630306 // indirect
-	golang.org/x/tools v0.1.12 // indirect
+	golang.org/x/tools v0.6.0 // indirect
```
Comment (maiqueb) [inline go.mod:60]: hm, this drop is very weird.

Comment (maiqueb) [reply go.mod:60]: @zshi-redhat ^

Comment (zshi-redhat) [reply go.mod:60]: @maiqueb  sorry for the late response.  It looks like it was dropped, but the minor version is increased from0.1 to 0.6.

Comment (maiqueb) [reply go.mod:60]: Oh I apologize, at a glance I mistakenly understood this as dropping from 0.12.0 to 0.6.0. My bad !!! 

Will approve !

**Comment by maiqueb** [comment]
@dougbtv / @zshi-redhat I cannot set the `backport-risk-assessed` label.

**Pull request merged**